src/main/java/com/comphenix/protocol/utility/MinecraftReflection.java

```diff
@@ -2274,6 +2273,10 @@ public static Class<?> getProfilePublicKeyClass() {
 		return getMinecraftClass("world.entity.player.ProfilePublicKey");
 	}
 
+    public static Class<?> getLoginSignatureClass() {
+        return getMinecraftClass("util.MinecraftEncryption").getClasses()[0];

```

Comment:
I don't love relying on the order here; does this class have any other identifying characteristics we could use?

Reply:
We could extract the type from the packet classes, but this seems unreliable in the future. There also the idea to scan all subclasses of Minecraft Reflection to get the correct constructor. Furthermore there is a class for signing chat messages. There we could use fuzzy reflection contract to extract the correct field with the expected constructor:

```suggestion
		Class<?> messageSigClass = getMinecraftClass("network.chat.MessageSignature", "MessageSignature");

		FuzzyClassContract signatureContract = FuzzyClassContract.newBuilder().
				constructor(FuzzyMethodContract.newBuilder().
						parameterCount(2).
						parameterSuperOf(Long.TYPE, 0).
						parameterSuperOf(byte[].class, 1).
						build()
				).build();

		FuzzyFieldContract fuzzyFieldContract = FuzzyFieldContract.newBuilder().
				typeMatches(signatureContract).
				build();

		return FuzzyReflection.fromClass(messageSigClass, true).getField(fuzzyFieldContract).getType();
```

Reply:
I would prefer using fuzzy reflection. It only needs to be run once, then the result gets cached